src/vm/gchelpers.cpp

```diff
@@ -1314,7 +1314,7 @@ void ErectWriteBarrierForMT(MethodTable **dst, MethodTable *ref)
 
     *dst = ref;
 
-#ifdef _DEBUG
+#ifdef WRITE_BARRIER_CHECK
```

Comment:
It is defined for internal checked builds. Otherwise the build break you have run into and that you have fixed by correcting this ifdef would show up in the internal checked builds as well.

`WRITE_BARRIER_CHECK` is defined in two places for the internal builds:
- [clr.props](https://github.com/dotnet/coreclr/blob/master/clr.props#L187) that defines it for `no opt` builds only. You have probably looked at just this one that confused you.
- [vm.settings](https://github.com/dotnet/coreclr/blob/master/src/vm/vm.settings#L42) that defines it for both internal debug and checked builds

The seconds one is super set of the first one. It would be nice you can delete the  redundant definition of `WRITE_BARRIER_CHECK` in `clr.props` and move the ``WRITE_BARRIER_CHECK` definition for CMake build to be under `src\vm` so that it is in sync with the internal build scripts.